lib/spack/spack/environment.py

```diff
@@ -326,6 +332,29 @@ def _write_yaml(data, str_or_file):
                      default_flow_style=False)
 
 
+def _eval_conditional(string):
+    """Evaluate conditional definitions using restricted variable scope."""
+    arch = architecture.Arch(
+        architecture.platform(), 'default_os', 'default_target')
+    valid_variables = {
```

Comment:
In documentation `spack_*` is mentioned, should this be changed?
If I use something like `when: arch == 'linux-centos7-ppc64le'` it appears to work.

Reply:
Yes, good catch. I changed them in the code but not the documentation.